### PR TITLE
#1-4Task. Andrey Tereshkin

### PR DESCRIPTION
**Pull request by Terem**

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -145,7 +149,7 @@ function removeFirstOccurrences(str, value) {
  *   '<a>' => 'a'
  */
 function unbracketTag(str) {
-    throw new Error('Not implemented');
+    return (str.replace(/>/, '')).replace(/</, '');
```
Comment (vramaniuk) [inline task/01-strings-tasks.js]: please, use one replace with regex inside

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -238,7 +248,10 @@ function encodeToRot13(str) {
  *   isString(new String('test')) => true
  */
 function isString(value) {
-    throw new Error('Not implemented');
+    if (typeof value === 'string' || value instanceof String)
+        return true;
+   
+    return false;
```
Comment (vramaniuk) [inline task/01-strings-tasks.js]: We can just `return (typeof (value) === 'string') || ( value instanceof String );` with the same result

Please, see https://javascript.info/logical-operators

File: task/02-numbers-tasks.js [hunk 3]
```diff
@@ -54,7 +54,11 @@ function getCicleCircumference(radius) {
  *  -3, 3  => 0
  */
 function getAverage(value1, value2) {
-    throw new Error('Not implemented');
+    let max = 1.7976931348623157e+308;
+    if (max < ((value1 + value2) / 2))
+        return 1.7976931348623157e+308;
+    return ((value1 + value2) / 2);
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: JS has Number.MAX_VALUE. 
Please use it in our logic

File: task/02-numbers-tasks.js [hunk 4]
```diff
@@ -73,7 +77,7 @@ function getAverage(value1, value2) {
  *   (-5,0) (10,-10) => 18.027756377319946
  */
 function getDistanceBetweenPoints(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    return Math.sqrt(Math.pow((x2 - x1), 2) + Math.pow((y2 - y1), 2));
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: Please, use Math.hypot

File: task/02-numbers-tasks.js [hunk 5]
```diff
@@ -159,8 +167,8 @@ function parseNumberFromString(value) {
  *   3,3,3   => 5.196152422706632
  *   1,2,3   => 3.741657386773941
  */
-function getParallelipidedDiagonal(a,b,c) {
-    throw new Error('Not implemented');
+function getParallelipidedDiagonal(a, b, c) {
+    return Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2));
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: Please, use Math.hypot

File: task/04-arrays-tasks.js [hunk 6]
```diff
@@ -23,7 +23,10 @@
  *    [0, 1, 2, 3, 4, 5], 5    => 5
  */
 function findElement(arr, value) {
-   throw new Error('Not implemented');
+   let res = arr.indexOf(value);
+   if (res !== -1)
+      return res;
+   return -1;
```
Comment (vramaniuk) [inline task/04-arrays-tasks.js]: Redundant logic
Please, just `return arr.indexOf(value)`

File: task/04-arrays-tasks.js [hunk 7]
```diff
@@ -102,7 +108,7 @@ function getArrayOfStrings(arr) {
  *    [ false, 0, NaN, '', undefined ]   => [ ]
  */
 function removeFalsyValues(arr) {
-   throw new Error('Not implemented');
+   return arr.filter((element) => { return (Boolean(element)) !== false });
```
Comment (vramaniuk) [inline task/04-arrays-tasks.js:108]: Redundant logic
`return arr.filter((element) => { return element  });` will also work
in this case element will be converted into boolean 
https://javascript.info/logical-operators

File: task/04-arrays-tasks.js [hunk 8]
```diff
@@ -304,9 +316,12 @@ function get3TopItems(arr) {
  *   [ 1, '2' ] => 1
  */
 function getPositivesCount(arr) {
-   throw new Error('Not implemented');
+   //return arr.filter(element => element > 0).length;
+   let arr_rez = [];
+   arr_rez = arr.filter(function (v) { return v > 0 & typeof v !== 'string'; });
```
Comment (vramaniuk) [inline task/04-arrays-tasks.js]: Please, use 'logical and'  `&&`

https://stackoverflow.com/questions/7310109/whats-the-difference-between-and-in-javascript#:~:text=To%20determine%20whether%20two%20boolean,to%20check%20the%20second%20value.

try in chrome console 
`console.log(true && 5)`
`console.log(true & 5)`

File: task/04-arrays-tasks.js [hunk 9]
```diff
@@ -353,7 +376,7 @@ function getItemsSum(arr) {
  *  [ null, undefined, NaN, false, 0, '' ]  => 6
  */
 function getFalsyValuesCount(arr) {
-   throw new Error('Not implemented');
+   return arr.filter(a => Boolean(a) === false).length;
```
Comment (vramaniuk) [inline task/04-arrays-tasks.js]: `arr.filter(a => !a).length`  it will work

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/04-arrays-tasks.js (hunk 9 above)
Comment (Terem) [reply task/04-arrays-tasks.js]: I made corrections and wanted to do new pull request but i could't. 
if i do something wrong let me know)

File: task/02-numbers-tasks.js [hunk 10]
```diff
@@ -54,7 +54,12 @@ function getCicleCircumference(radius) {
  *  -3, 3  => 0
  */
 function getAverage(value1, value2) {
-    throw new Error('Not implemented');
+    //let max = 1.7976931348623157e+308;
+    let max = Number.MAX_VALUE;
+    if (max < ((value1 + value2) / 2))
+        return 1.7976931348623157e+308;
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: Why haven't you used Number.MAX_VALUE here?

File: task/02-numbers-tasks.js [hunk 11]
```diff
@@ -73,7 +78,8 @@ function getAverage(value1, value2) {
  *   (-5,0) (10,-10) => 18.027756377319946
  */
 function getDistanceBetweenPoints(x1, y1, x2, y2) {
-    throw new Error('Not implemented');
+    //return Math.sqrt(Math.pow((x2 - x1), 2) + Math.pow((y2 - y1), 2));
+    return Math.hypot((x2-x1),(y2-y1));
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: space between parameters

File: task/02-numbers-tasks.js [hunk 12]
```diff
@@ -159,8 +169,9 @@ function parseNumberFromString(value) {
  *   3,3,3   => 5.196152422706632
  *   1,2,3   => 3.741657386773941
  */
-function getParallelipidedDiagonal(a,b,c) {
-    throw new Error('Not implemented');
+function getParallelipidedDiagonal(a, b, c) {
+    //return Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2));
+    return Math.hypot(a,b,c);
```
Comment (vramaniuk) [inline task/02-numbers-tasks.js]: space between parameters

Please, try to follow these rules https://javascript.info/coding-style

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/02-numbers-tasks.js (hunk 10 above)
Comment (Terem) [reply task/02-numbers-tasks.js]: very sorry i didn't notice

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**